crates/biome_cli/src/reporter/mod.rs

```diff
@@ -28,58 +30,24 @@ pub struct TraversalSummary {
 }
 
 /// When using this trait, the type that implements this trait is the one that holds the read-only information to pass around
-pub trait Reporter {
+pub trait Reporter: Sized {
     /// Writes the summary using the underling visitor
-    fn write(&mut self, visitor: &mut dyn ReporterVisitor) -> io::Result<()>;
+    fn write(self, visitor: &mut dyn ReporterVisitor) -> io::Result<()>;
 }
 
 /// When using this trait, the type that implements this trait is the one that will **write** the data, ideally inside a buffer
 pub trait ReporterVisitor {
     /// Writes the summary in the underling writer
     fn report_summary(
         &mut self,
-        traversal_mode: &Execution,
-        summary: &TraversalSummary,
+        execution: &Execution,
+        summary: TraversalSummary,
     ) -> io::Result<()>;
 
     /// Writes a diagnostics
-    fn report_diagnostics(&mut self, payload: &DiagnosticsPayload) -> io::Result<()>;
-}
-
-/// This function reports the result of a traversal
-pub(crate) fn report<R, V>(
```

Comment:
This function was causing compiler issues, it's not needed. The error was around `Reporter` not having a size that could not be known at compile time